src/main/java/org/opensearch/security/configuration/SecurityConfigVersionsLoader.java

```
@@ -0,0 +1,193 @@
+        }
+
+        if (failure[0] != null) {
+            throw new RuntimeException("Failed to load latest config version", failure[0]);
+        }
+
+        return result.get();
+    }
+
+    public SecurityConfigVersionDocument loadFullDocument() {
+        final AtomicReference<SecurityConfigVersionDocument> result = new AtomicReference<>();
+
+        final Exception[] error = new Exception[1];
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        getSecurityConfigVersionDocAsync(new ActionListener<>() {
+            @Override
+            public void onResponse(SecurityConfigVersionDocument doc) {
+                result.set(doc);
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                error[0] = e;
+                latch.countDown();
+            }
+        });
+
+        try {
+            if (!latch.await(10, TimeUnit.SECONDS)) {
+                throw new RuntimeException("Timeout while loading full config version document");
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Interrupted while loading full config version document", e);
+        }
+
+        if (error[0] != null) {
+            throw new RuntimeException("Failed to load full config version document", error[0]);
+        }
+
+        return result.get() != null ? result.get() : new SecurityConfigVersionDocument();
+
+    }
+
+    public static <T> void sortVersionsById(List<SecurityConfigVersionDocument.Version<?>> versions) {
+        versions.sort((v1, v2) -> {
+            try {
+                int n1 = Integer.parseInt(v1.getVersion_id().substring(1));
+                int n2 = Integer.parseInt(v2.getVersion_id().substring(1));
+                return Integer.compare(n1, n2);
+            } catch (Exception e) {
+                log.warn("Invalid version_id format", e);
```

Is it okay to silently ignore this ?